test/integration/client-side-encryption/client_side_encryption.test.ts

```diff
@@ -0,0 +1,251 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+
+// eslint-disable-next-line @typescript-eslint/no-restricted-imports
+import { StateMachine } from '../../../src/client-side-encryption/state_machine';
+import { BSON, Connection, CSOTTimeoutContext, MongoOperationTimeoutError } from '../../mongodb';
+import { type FailPoint, sleep } from '../../tools/utils';
+
+describe('Client-Side Encryption (Integration)', function () {
+  describe('CSOT', function () {
+    describe('Auto encryption', function () {
+      let setupClient;
+
+      beforeEach(async function () {
+        setupClient = this.configuration.newClient();
+        await setupClient
+          .db()
+          .admin()
+          .command({
+            configureFailPoint: 'failCommand',
+            mode: 'alwaysOn',
+            data: {
+              failCommands: ['aggregate'],
+              blockConnection: true,
+              blockTimeMS: 2000
+            }
+          } as FailPoint);
+      });
+
+      afterEach(async function () {
+        await setupClient
+          .db()
+          .admin()
+          .command({
+            configureFailPoint: 'failCommand',
+            mode: 'off'
+          } as FailPoint);
+        await setupClient.close();
+      });
+
+      context('when client is provided timeoutMS and command hangs', function () {
+        let encryptedClient;
```

Comment:
(just a suggestion)


```suggestion
        let encryptedClient: MongoClient;
```

Then you get red underlines and type hinting when writing tests using the client.